src/BlockTypes/ProductQuery.php

```diff
@@ -80,41 +83,107 @@ public function get_query_by_attributes( $query ) {
 			'posts_per_page' => $query['posts_per_page'],
 			'orderby'        => $query['orderby'],
 			'order'          => $query['order'],
+			// Ignoring the warning of not using meta queries.
+			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+			'meta_query'     => array(),
+		);
+
+		$queries_attributes = $this->get_queries_by_attributes( $variation_props );
+		$queries_filters    = $this->get_queries_by_applied_filters();
+
+		return array_reduce(
+			array_merge(
+				$queries_attributes,
+				$queries_filters
+			),
+			function( $acc, $query ) {
+						// Ignoring the warning of not using meta queries.
+						// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+						$acc['meta_query'] = isset( $query['meta_query'] ) ? array_merge( $acc['meta_query'], array( $query['meta_query'] ) ) : $acc['meta_query'];
+						return $acc;
+			},
```

Comment:
If we go with the `post__in` approach, this `array_reduce()` is ignoring that property, right? Because we are only taking the `meta_query` value from `$queries_attributes` and `$queries_filters`, but not `post__in`. I might be missing something, though. :thinking: